packages/pieces/src/framework/property/property.ts

```diff
@@ -0,0 +1,54 @@
+import {
+	BasePropertySchema,
+	CheckboxProperty,
+	LongTextProperty,
+	NumberProperty,
+	PropertyType,
+	SecretTextProperty,
+	ShortTextProperty,
+} from "./base-prop";
+import { CustomAuthProperty, CustomAuthPropertySchema } from "./custom-auth-prop";
+import { DropdownProperty, DropdownPropertySchema } from "./dropdown-prop";
+import { OAuth2Property, OAuth2PropertySchema } from "./oauth-prop";
+
+export interface PieceProperty {
+	[name: string]: ShortTextProperty
+		| LongTextProperty
+		| OAuth2Property
+		| CheckboxProperty
+		| DropdownProperty<any>
+		| NumberProperty
+		| SecretTextProperty
+		| CustomAuthProperty;
+}
+
+export type StaticPropsValue<T extends PieceProperty> = {
+	[P in keyof T]: T[P]['valueSchema'];
+}
+
+export const Property = {
+	ShortText(request: BasePropertySchema): ShortTextProperty {
+		return {...request, valueSchema: undefined, type: PropertyType.SHORT_TEXT};
+	},
+	Checkbox(request: BasePropertySchema): CheckboxProperty {
+		return {...request, valueSchema: undefined, type: PropertyType.CHECKBOX};
+	},
+	LongText(request: BasePropertySchema): LongTextProperty {
+		return {...request, valueSchema: undefined, type: PropertyType.LONG_TEXT};
+	},
+	Number(request: BasePropertySchema): NumberProperty {
+		return {...request, valueSchema: undefined, type: PropertyType.NUMBER};
+	},
+	OAuth2(request: OAuth2PropertySchema): OAuth2Property {
+		return {...request, valueSchema: undefined, type: PropertyType.OAUTH2};
+	},
+	Dropdown<T>(request: DropdownPropertySchema<T>): DropdownProperty<T> {
+		return {...request, valueSchema: undefined, type: PropertyType.DROPDOWN};
+	},
+	ApiKey(request: BasePropertySchema): SecretTextProperty {
```

Comment:
Is this intended? to be api key not secret key?

Reply:
ouch